src/services/componentService.ts

```diff
@@ -201,6 +209,75 @@ export const fetchAndStoreComponent = async (
   }
 };
 
+/**
+ * Fetch and store a single component by spec, text, or URL (in that order).
+ * Returns the populated ComponentSpec or null.
+ */
+export const fetchAndStoreComponent = async (
+  component: ComponentReference,
+): Promise<ComponentSpec | null> => {
+  try {
+    const id = `component-${Date.now()}-${Math.random().toString(36).substring(2, 9)}`;
+    const createdAt = Date.now();
+    const updatedAt = Date.now();
+
+    let text: string | undefined;
+    let spec: ComponentSpec | null | undefined;
+
+    // 1. If spec is present, use it and serialize to text
```

Comment:
The comment seems incorrect (does not correspond to the code).
Also, serializing spec to text is pretty bad and should only be done at last resort (when it's not possible to get text (e.g. from text or URL)).